src/types.ts

```diff
@@ -16,8 +16,10 @@ export interface ProxyTargetDetailed {
 }
 
 type ProxyTarget = ProxyTargetUrl | ProxyTargetDetailed;
+export type NormalizedProxyTarget = Exclude<ProxyTarget, string>;
 
-type ProxyTargetUrl = string | Partial<URL>;
+type ProxyTargetUrl = string | URL;
```

Comment:
I changed `Partial<URL>` to `URL` here as I wasn't sure how to handle objects like `{ pathname: '/foo' }` and didn't make sense to me.